### PR TITLE
add MonitoringResultFetcher to fetch monitoring

### PR DESCRIPTION
**Pull request by Yifeng-Sigma**
Multiplex PR: https://github.com/sigmacomputing/mono-go/pull/1040

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -756,3 +756,19 @@ func (sc *snowflakeConn) FetchResult(ctx context.Context, qid string) (driver.Ro
 type ResultFetcher interface {
 	FetchResult(ctx context.Context, qid string) (driver.Rows, error)
 }
+
+func (sc *snowflakeConn) FetchMonitoringResult() (
```
Comment (mtoader) [inline connection.go:760]: weird formatting of parameters here

File: connection.go [hunk 2]
```diff
@@ -756,3 +756,19 @@ func (sc *snowflakeConn) FetchResult(ctx context.Context, qid string) (driver.Ro
...
+	*monitoringResult, error) {
+	if sc.rest == nil {
+		return nil, driver.ErrBadConn
+	}
+
+	// set the fake runtime just to bypass fast query
+	monitoringResult := mkMonitoringFetcher(sc, sc.QueryID, time.Minute*10)
```
Comment (mtoader) [inline connection.go:767]: Are there any concurrency issues with using this? I remember the first time @mhseiden added the shim to fetch monitoring data we needed up with concurrency issue later one (@agam looked into it if I remember correctly).

Comment (agam) [reply connection.go:766]: This is the fix for a data-race I had made earlier: https://github.com/sigmacomputing/gosnowflake/pull/32/commits/0540a663df0afd11f02fdc0a122e3ac9513a53f7

The race was between the monitoring-fetcher and the _connection_. It's independent of whether or not we have a `rows` object, so we should be fine here.

File: connection.go [hunk 3]
```diff
@@ -756,3 +756,17 @@ func (sc *snowflakeConn) FetchResult(ctx context.Context, qid string) (driver.Ro
 type ResultFetcher interface {
 	FetchResult(ctx context.Context, qid string) (driver.Rows, error)
 }
+
+func (sc *snowflakeConn) FetchMonitoringResult() (*monitoringResult, error) {
```
[the remark this answers is not in the document]
Comment (Yifeng-Sigma) [reply connection.go:760]: > * Do we need to expose the `monitoringResult` type?
`FetchMonitoringResult() ` and `func (m *monitoringResult) Monitoring(wait time.Duration) *QueryMonitoringData ` are exported, and we don't need to create the `monitoringResult`. so it seems fine. ref: https://stackoverflow.com/questions/37952354/golang-public-method-to-private-struct-is-this-have-any-use-case

[the remark this answers is not in the document]
Comment (Yifeng-Sigma) [reply connection.go:760]: > * The `QueryMonitoringData` is already public, maybe we need to replicate what `SnowflakeResult::Monitoring` does, and return _that_ here?

I think we can just use this exported `Monitoring() `on `monitoringResult` https://github.com/sigmacomputing/gosnowflake/blob/master/result.go#L93

**Review by mtoader — APPROVED** [review]

**Pull request merged**